discord/utils.py

```diff
@@ -931,6 +934,50 @@ def escape_mentions(text: str) -> str:
     """
     return re.sub(r"@(everyone|here|[!&]?[0-9]{17,20})", "@\u200b\\1", text)
 
+def raw_mentions(text: str) -> List[int]:
+    """Returns a list of user IDs matching ``<@user_id>`` in the string.
+
+    Parameters
+    -----------
+    text: :class:`str`
+        The string to get user mentions from.
+
+    Returns
+    --------
+    List[:class:`int`]
+        A list of user IDs found in the string.
+    """
+    return [int(x) for x in re.findall(r"<@!?([0-9]{15,20})>", text)]
```

Comment:
if we want to get technical, this fails for clyde `<@1>`. Maybe just allow any length. Also, the max length of a 64 bit unsigned is 19. Maybe same for other functions for parity?
```suggestion
    return [int(x) for x in re.findall(r"<@!?([0-9]{1,19})>", text)]
```

Reply:
Interesting, didn't even know Clyde mentions were a thing (though as far as I can tell you can't do it manually?)

I stuck with `{15,20}` since that was the standard across all the `raw_` functions, and functionally it'd keep matches in a reasonable range (e.g. not matching something randomly small like `<@0123794>`), but if there's valid usage for allowing a lower range then fair enough.

> The max length of a 64 bit unsigned is 19

Is it not 19 for signed and 20 for unsigned? Though we won't hit 20 digits for several decades...

Reply:
> though as far as I can tell you can't do it manually?

You can mention clyde manually, you just have to have it cached (in this example I invoked a command that clyde responded to).

https://user-images.githubusercontent.com/71356958/193412523-552158dc-0005-42c9-9916-cdb7e22043e6.mov

> Is it not 19 for signed and 20 for unsigned?

My bad, that's correct.

> I stuck with `{15,20}` since that was the standard across all the `raw_` functions

TBH I think we should change that to `{1,20}` on all of them. As shown by clyde, we should assume that even invalid IDs are parsed as mentions. In addition, they're always parsed as mentions by the client. TBH, we could even use `[0-9]+` as that's what the client appears to use.

Reply:
Indeed, the client treats it as `[0-9+]` so it would be more appropriate.
Should this adjustment also be committed to the original functions from `Message`, or leave them as is?

Reply:
Yes, but in a separate PR